asset-transfer-secured-agreement/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,427 @@
+
+        // Read the public details by  org1.
+        await readAsset(assetKey, mspIdOrg2, contractOrg1,mspIdOrg1);
+
+        // Read the public details by  org2.
+        await readAsset(assetKey, mspIdOrg2, contractOrg2,mspIdOrg2);
+
+        // Org2 should be able to read the private data details of this asset
+        await readPrivateAssetAfterTransfer(assetKey, mspIdOrg2, contractOrg2);
+
+        // Org1 should not be able to read the private data details of this asset,expected to fail
+        await readPrivateAssetAfterTransfer(assetKey, mspIdOrg1, contractOrg1);
+
+        // This is an update to the public state and requires only the owner to endorse.
+        // Org2 wants to indicate that the items is no longer for sale
+        await changePublicDescriptionAfterTransfer(assetKey,mspIdOrg2,`Asset ${assetKey} owned by ${mspIdOrg2} is NOT for sale`,contractOrg2);
+
+        // Read the public details by  org1.
+        await readAsset(assetKey, mspIdOrg2, contractOrg1,mspIdOrg1);
+
+        // Read the public details by  org2.
+        await readAsset(assetKey, mspIdOrg2, contractOrg2,mspIdOrg2);
+
+    } finally {
+        gatewayOrg1.close();
+        gatewayOrg2.close();
+        clientOrg1.close();
+        clientOrg2.close();
+    }
+}
+
+main().catch(error => {
+    console.error('******** FAILED to run the application:', error);
+    process.exitCode = 1;
+});
+
+async function createAsset(contract:Contract,org:string):Promise<void> {
+    const asset_properties = {
+        object_type: 'asset_properties',
+        asset_id: assetKey,
+        color: 'blue',
+        size: 35,
+        salt: Buffer.from(now.toString()).toString('hex')
+    };
+    console.log(`${GREEN}--> Submit Transaction: CreateAsset, ${assetKey} as ${org} - endorsed by Org1${RESET}`);
+
+    await contract.submit('CreateAsset', {
+        arguments:[assetKey,`Asset ${assetKey} owned by ${org} is not for sale`],
+        transientData: { asset_properties: JSON.stringify(asset_properties) },
+    });
+
+    console.log(`*** Result: committed, asset ${assetKey} is owned by Org1`);
+}
+
+async function readAsset(assetKey:string, ownerOrg:string, contract:Contract,org:string) {
+    console.log(`${GREEN}--> Evaluate Transactions: ReadAsset as ${org}, - ${assetKey} should be owned by ${ownerOrg}${RESET}`);
+
+    const resultBytes = await contract.evaluateTransaction('ReadAsset', assetKey);
+
+    const resultString = utf8Decoder.decode(resultBytes);
+    const result = JSON.parse(resultString);
+
+    if (result) {
+        if (result.ownerOrg === ownerOrg) {
+            console.log(`*** Result from ${org} - asset ${result.assetID} owned by ${result.ownerOrg} DESC:${result.publicDescription}`);
+        } else {
+            console.log(`${RED}*** Failed owner check from ${org} - asset ${result.assetID} owned by ${result.ownerOrg} DESC:${result.publicDescription}${RESET}`);
+        }
+    }
```

Comment:
... if result is null?

Reply:
As Matthew points out, you probably want a better check. Something like:
```typescript
if (result?.ownerOrg === ownerOrg) {
    console.log(`*** Result from ${org} - asset ${result.assetID} owned by ${result.ownerOrg} DESC:${result.publicDescription}`);
} else {
    console.log(`${RED}*** Failed owner check from ${org} - asset ${result.assetID} owned by ${result.ownerOrg} DESC:${result.publicDescription}${RESET}`);
}
```
You don't need the surrounding `if (result) {}`.

Although there is another code smell here, which is that you're dealing with `any` type values. I wonder if it would be worth defining an interface that describes the shape of the objects you expect back from `JSON.parse()`?